packages/block-editor/src/components/block-canvas/index.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Internal dependencies
+ */
+import BlockList from '../block-list';
+import EditorStyles from '../editor-styles';
+import Iframe from '../iframe';
+import WritingFlow from '../writing-flow';
+import { useMouseMoveTypingReset } from '../observe-typing';
+
+function BlockCanvas( {
+	shouldIframe = true,
+	children = <BlockList />,
+	styles,
+	contentRef,
+	iframeProps,
+} ) {
+	const resetTypingRef = useMouseMoveTypingReset();
```

Comment:
I don't know if the call useMouseMoveTypingReset is expensive in terms of performance. But suppose it has some impact given that it is not needed when shouldIframe is false. In that case, we may consider having an inner internal component wrapping the Iframe and making the useMouseMoveTypingReset call, or even call useMouseMoveTypingReset inside the iframe if it is always needed.

Reply:
This hook does nothing really unless you pass it to an element, so it's fine.